tests/core/test_model.py

```diff
@@ -4248,6 +4250,64 @@ def test_model_session_properties(sushi_context):
         "warehouse": "test_warehouse",
     }
 
+    model = load_sql_based_model(
+        d.parse(
+            """
+        MODEL (
+            name test_schema.test_model,
+            session_properties (
+                'query_label' = [
+                    ('key1', 'value1'),
+                    ('key2', 'value2')
+                ]
+            )
+        );
+        SELECT a FROM tbl;
+        """,
+            default_dialect="bigquery",
+        )
+    )
+    assert model.session_properties == {
+        "query_label": exp.Array(
+            expressions=[
+                exp.Tuple(
+                    expressions=[
+                        exp.Literal.string("key1"),
+                        exp.Literal.string("value1"),
+                    ]
+                ),
+                exp.Tuple(
+                    expressions=[
+                        exp.Literal.string("key2"),
+                        exp.Literal.string("value2"),
+                    ]
+                ),
+            ]
+        )
+    }
+
+    model = load_sql_based_model(
+        d.parse(
+            """
+        MODEL (
+            name test_schema.test_model,
+            session_properties (
+                'query_label' = (
+                    ('key1', 'value1')
+                )
+            )
+        );
+        SELECT a FROM tbl;
+        """,
+            default_dialect="bigquery",
+        )
+    )
+    assert model.session_properties == {
```

Comment:
Can we please also add a few test cases that fail the validation

Reply:
Good point. I added them